Scripts/Tracking/Velocity/XRNodeVelocityEstimator.cs

```diff
@@ -0,0 +1,59 @@
+﻿namespace VRTK.Core.Tracking.Velocity
+{
+    using System.Collections.Generic;
+    using UnityEngine;
+    using UnityEngine.XR;
+
+    /// <summary>
+    /// Retrieves the velocity estimations for the specified <see cref="XRNode"/> device.
+    /// </summary>
+    public class XRNodeVelocityEstimator : VelocityTracker
+    {
+        /// <summary>
+        /// The device to query velocity estimations for.
+        /// </summary>
+        [Tooltip("The device to query velocity estimations for.")]
+        public XRNode device = XRNode.LeftHand;
+
+        /// <inheritdoc />
+        public override bool IsActive()
+        {
+            return isActiveAndEnabled;
+        }
+
+        /// <inheritdoc />
+        protected override Vector3 DoGetVelocity()
+        {
+            Vector3 result = Vector3.zero;
+            GetNodeState().TryGetVelocity(out result);
+            return result;
+        }
+
+        /// <inheritdoc />
+        protected override Vector3 DoGetAngularVelocity()
+        {
+            Vector3 result = Vector3.zero;
+            GetNodeState().TryGetAngularVelocity(out result);
+            return result;
+        }
+
+        /// <summary>
+        /// Gets the state of the node associated with the specified <see cref="XRNode"/> device.
+        /// </summary>
+        /// <returns>The associated node state.</returns>
+        protected virtual XRNodeState GetNodeState()
+        {
+            List<XRNodeState> nodesStates = new List<XRNodeState>();
```

Comment:
Make this a `protected` instance variable that we can reuse. This makes this API usage almost always allocation-free.